source/E2E.Tests/Services/BesiegerCamps/BesiegerCampCollectionTests.cs

```diff
@@ -0,0 +1,185 @@
+﻿using E2E.Tests.Environment;
+using E2E.Tests.Environment.Instance;
+using E2E.Tests.Util;
+using GameInterface.Services.BesiegerCamps.Patches;
+using HarmonyLib;
+using System.Reflection;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Siege;
+using Xunit.Abstractions;
+
+namespace E2E.Tests.Services.BesiegerCamps
+{
+    public class BesiegerCampCollectionTests : IDisposable
+    {
+        private List<MethodBase> disabledMethods = new();
+
+        private E2ETestEnvironment TestEnvironment { get; }
+
+        private EnvironmentInstance Server => TestEnvironment.Server;
+        private IEnumerable<EnvironmentInstance> Clients => TestEnvironment.Clients;
+
+        private IEnumerable<EnvironmentInstance> AllEnvironmentInstances => Clients.Append(Server);
+
+        public BesiegerCampCollectionTests(ITestOutputHelper output)
+        {
+            TestEnvironment = new E2ETestEnvironment(output);
+            DisableMethods();
```

Comment:
Space in typeof looks weird to me, was "typeof (MobileParty)", just being pedantic 
```suggestion
            disabledMethods = new List<MethodBase> {
                AccessTools.Method(typeof(MobileParty), nameof(MobileParty.OnPartyJoinedSiegeInternal)),
                AccessTools.Method(typeof(BesiegerCamp), nameof(BesiegerCamp.SetSiegeCampPartyPosition)),
                AccessTools.Method(typeof(BesiegerCamp), nameof(BesiegerCamp.InitializeSiegeEventSide))
            };
```

Reply:
no idea where those spaces came from